src/qibo/backends/numpy.py

```diff
@@ -592,8 +594,13 @@ def calculate_symbolic_density_matrix(
                 return terms
         return terms
 
+    def dimensions(self, x):
+        return x.ndim
+
     def _order_probabilities(self, probs, qubits, nqubits):
         """Arrange probabilities according to the given ``qubits`` ordering."""
+        if self.dimensions(probs) == 0:
+            return probs
```

Comment:
I guess this is where the new method is needed. It is better to just redefine `_order_probabilities` in the `torch` class